.github/workflows/rust.yml

```
@@ -0,0 +1,36 @@
+name: Rust
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    
+    - name: Setup Rust
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+        override: true
+
+    
+    - name: Build
+      run: cargo build --verbose
+
+    - name: Run tests
+      run: cargo test --locked --verbose
+
+    - name: Check formatting
+      run: cargo fmt --check
+
+    - name: Lint with Clippy
+      run: cargo clippy -- -D warni
```

Also, it looks like this isn't properly denying warnings due to typo:
```suggestion
      run: cargo clippy -- -D warnings
```